src/replay/commit.zig

```diff
@@ -63,7 +76,24 @@ pub const Committer = struct {
             }
 
             switch (tx_result) {
-                .executed => |exec| rent_collected += exec.loaded_accounts.rent_collected,
+                .executed => |exec| {
+                    rent_collected += exec.loaded_accounts.rent_collected;
+                    // Skip non successful or non vote transactions.
+                    if (exec.executed_transaction.err == null and
+                        isSimpleVoteTransaction(transaction.transaction))
+                    {
+                        if (vote_listener.vote_parser.parseSanitizedVoteTransaction(
+                            allocator,
+                            transaction,
+                        ) catch null) |parsed| {
```

Comment:
Why are we treating this OOM as null? It should just propagate with `try` like all other instances (like the calls to `putAccount`).

Reply:
I think it boils down to the question: if an attempt to send out a vote tx down to the vote listener fails, should the remaining operations in the commit path be aborted? The current stance is, the vote tx should just be skipped, allowing the rest of the commit processing to continue.

Reply:
That isn't reflected in the other parts of this function which simply propagate `try` as a normal error, which seems like it should be the default stance, unless we're going to be handling OOM in some specific way. Parsing errors we can skip, but if there's an OOM, we want to fail, otherwise we're just deferring the error to a future allocation, which will be further removed from the initial actual OOM.